op-e2e/helper.go

```diff
@@ -14,6 +17,36 @@ type testopts struct {
 	executor uint64
 }
 
+var (
+	testBaseExp      = regexp.MustCompile("^Test[^/]+")
+	testRegistryInst = testRegistry{
+		tests: map[string]uint64{},
+	}
+)
+
+type testRegistry struct {
+	tests map[string]uint64
+	mutex sync.Mutex
+}
+
+// registerTest ensures that if a test is assigned to an executor, that its
+// parent test is assigned to the same executor, or not assigned to an executor
+// at all
+func (tr *testRegistry) registerTest(t e2eutils.TestingBase, name string, executor uint64) {
+	baseName := testBaseExp.FindString(name)
+	tr.mutex.Lock()
+	defer tr.mutex.Unlock()
+	if baseExecutor, ok := tr.tests[baseName]; ok {
+		require.Equal(
+			t, baseExecutor, executor,
+			"base test for %s executes only on %d but %s requested to execute on %d",
+			baseName, baseExecutor, name, executor,
+		)
```

Comment:
If I'm understanding correctly, this will cause subtests to fail if they wind up scheduled on different executors.  That means tests will potentially start failing if we change the number of executors in use because they were previously allocated to the same executor but aren't anymore.  It also means we can't use `InitParallel` on subtests and so all the subtests within a parent run sequentially.

I was thinking we'd just always run subtests if they are invoked so it's only the top level tests that are spread across executors. Does that make sense to you?

Reply:
Put up https://github.com/ethereum-optimism/optimism/pull/9593 which I think does this (CI is running and will hopefully identify this failing withdrawal test).

Reply:
Yeah, that behavior makes more sense.  Splitting subtests across runners felt strange to me, but I thought that maybe there was a good reason for it.  Feel free to close this one in favor of #9593